test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -68,9 +68,9 @@
              ;; We evaluate the first element from lazy-seq just to check if DB query was successful or not
              ;; so we have to ensure the first element and the rest have been realized, not just the first
              ;; element on its own.
-             (reset! before-slurp? (and (realized? result-set) (realized? (rest result-set))))
-             (reset! results (vec result-set))
-             (reset! after-slurp? (and (realized? result-set) (realized? (rest result-set))))))
+             (reset! before-slurp? (realized? result-set))
+             (reset! results (into [] result-set))
+             (reset! after-slurp? (realized? result-set))))
```

Comment:
So I'm guessing from your hipchat comments that this won't be sufficient for testing this @senior?